darts/utils/torch.py

```diff
@@ -112,3 +112,34 @@ def decorator(self, *args, **kwargs) -> T:
             return decorated(self, *args, **kwargs)
 
     return decorator
+
+
+class TemporalBatchNorm1d(nn.Module):
+    def __init__(self, feature_size) -> None:
+        super().__init__()
+        self.norm = nn.BatchNorm1d(feature_size)
+
+    def forward(self, input):
+        input = self._reshape_input(input)  # Reshape N L C -> N C L
```

Comment:
This line is more about swapping axes that reshaping, I would instead use the corresponding torch function:
```suggestion
        # Reshape N L C -> N C L
        input = input.swapaxes(1,2)  
```